plan/physical_plan_builder.go

```diff
@@ -61,6 +62,14 @@ func (p *DataSource) convert2TableScan(prop *requiredProperty) (*physicalPlanInf
 		ts.readOnly = true
 	}
 
+	infoSchema, ok := getInfoSchema(p.ctx)
```

Comment:
Maybe you misunderstand what I mean:
During plan build, it's unsafe to get and use information schema from ctx again.

Bear in mind that any time you call `txn.InfoSchema`, it may get a schema different from previous one. Maybe the schema added one column, or removed one column, whatever.

Building plan involves many small steps :

```
func plan_build() {
    schema1 = ctx.InfoSchema
    build_plan_step1(schema1)   //  in this step, a table has 4 columns, plan alloc a array[4]
    schema2 = ctx.InfoSchema
    build_plan_step2(schema2)   //  in this step, the same table has 5 columns, plan may visit a array offset[4] and panic!
}
```